### PR TITLE
Update test server config

### PR DESCRIPTION
**Pull request by ricellis**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

Change the test server configuration and allow for overriding the token server used by tests.

### 1. Steps to reproduce and the simplest code sample possible to demonstrate the issue

Run Jenkins tests with a custom IAM token server.

### 2. What you expected to happen

Tests to pass.

### 3. What actually happened

Test failures.

## Approach

1. Add `TEST_IAM_TOKEN_URL` environment variable
    - Update `Jenkinsfile`
    - Update `cloudant-client/build.gradle` 
2. Fix tests to work with above (see "Testing" below)
3. Added new env vars and cred IDs to `Jenkinsfile`

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- Modified existing tests to support changing IAM token server.
    - Map `TEST_IAM_TOKEN_URL` env var into `com.cloudant.client.iamserver` system property for test JVM
    - Move test env var mapping to apply to all gradle tasks of type `Test`
    - Add `MockIamWebServerExtension` to temporarily change `com.cloudant.client.iamserver` system property during tests using a mock IAM server, replacing the configured value after tests. Consume this extension in these tests:
        - cloudant-client/src/test/java/com/cloudant/tests/CloudFoundryServiceTest.java
        - cloudant-client/src/test/java/com/cloudant/tests/HttpIamTest.java
        - cloudant-client/src/test/java/com/cloudant/tests/SessionInterceptorExpiryTests.java

IAM testing normally only runs on the `master` branch - I temporarily enabled it for this branch and observed 7 failures relating to the tuning configuration of the new test server. They should resolve when the server configuration is updated.

## Monitoring and Logging

- "No change"

### CODE REVIEW

**Review by emlaver — COMMENTED** [review]
Looks good.  
Spotbugs task is throwing an error of `No such DSL method 'spotBugs' found among steps`

File: Jenkinsfile [hunk 1]
```diff
@@ -49,8 +55,8 @@ stage('Build') {
 
 stage('QA') {
     // Define the matrix environments
-    def CLOUDANT_ENV = ['DB_HTTP=https', 'DB_HOST=clientlibs-test.cloudant.com', 'DB_PORT=443', 'DB_IGNORE_COMPACTION=true', 'CREDS_ID=clientlibs-test']
-    def CLOUDANT_IAM_ENV = ['DB_HTTP=https', 'DB_HOST=clientlibs-test.cloudant.com', 'DB_PORT=443', 'DB_IGNORE_COMPACTION=true', 'CREDS_ID=clientlibs-test-iam']
+    def CLOUDANT_ENV = ['DB_HTTP=https', "DB_HOST=${SDKS_TEST_SERVER_HOST}", 'DB_PORT=443', 'DB_IGNORE_COMPACTION=true', 'CREDS_ID=testServerLegacy']
```
Comment (mojito317) [inline Jenkinsfile:58]: Nit:
```suggestion
    def CLOUDANT_ENV = ['DB_HTTP=https', 'DB_HOST=$SDKS_TEST_SERVER_HOST', 'DB_PORT=443', 'DB_IGNORE_COMPACTION=true', 'CREDS_ID=testServerLegacy']
```

File: Jenkinsfile [hunk 2]
```diff
@@ -49,8 +55,8 @@ stage('Build') {
...
+    def CLOUDANT_IAM_ENV = ['DB_HTTP=https', "DB_HOST=${SDKS_TEST_SERVER_HOST}", 'DB_PORT=443', 'DB_IGNORE_COMPACTION=true', 'CREDS_ID=testServerIamApiKey', "TEST_IAM_TOKEN_URL=${SDKS_TEST_IAM_URL}"]
```
Comment (mojito317) [inline Jenkinsfile:59]: Nit:
```suggestion
    def CLOUDANT_IAM_ENV = ['DB_HTTP=https', 'DB_HOST=$SDKS_TEST_SERVER_HOST', 'DB_PORT=443', 'DB_IGNORE_COMPACTION=true', 'CREDS_ID=testServerIamApiKey', "TEST_IAM_TOKEN_URL=${SDKS_TEST_IAM_URL}"]
```

Comment (ricellis) [reply Jenkinsfile]: Done in 6c4397a9 including its friend `TEST_IAM_TOKEN_URL` as well

File: Jenkinsfile (hunk 1 above)
Comment (ricellis) [reply Jenkinsfile:58]: Actually this doesn't work after all so I reverted it during my rebase.

File: Jenkinsfile (hunk 2 above)
Comment (ricellis) [reply Jenkinsfile:59]: Sorry as per above, it didn't work, reverted.

Comment (mojito317) [reply Jenkinsfile:59]: Ack. Thanks for trying it out though!

**Review by mojito317 — APPROVED** [review]

**Comment by ricellis** [comment]
@emlaver @mojito317 sorry I missed a test on my first run through this, I added another fix in 497d7882 if you want to take a look.

**Comment by ricellis** [comment]
@emlaver the `spotBugs` things is weird, but AFAICT isn't related to this change.

**Comment by ricellis** [comment]
@emlaver good news, `spotBugs` issue is resolved

**Review by emlaver — APPROVED** [review]

**Pull request merged**